test/InputStates-mocha.js

```diff
@@ -0,0 +1,261 @@
+        context.should.not.be.undefined;
+        context.process.should.exist;
+        context.process.argv.should.exist;
+        context.process.env.should.exist;
+
+        // Verify the correct input state is there
+        var myInputState = InputStates( inportName,
+                                        sourceNodeName,
+                                        sourceNodePort );
+        myInputState.should.be.an('object');
+        myInputState.should.have.all.keys('data', 'lm');
+        myInputState.data.should.equal(testData);
+        myInputState.lm.should.equal(testLm);
+    });
+
+    it("should change state on a noflo component", function() {
+        return Promise.resolve({
+            inPorts:{
+                'input':{
+                     description: "test input port",
+                }
+            },
+            outPorts:{
+                'output':{
+                     description: "test output port",
+                }
+            }
+        }).then(componentFactory).then(test.createComponent).then(function(component){
+            return new Promise(function(callback){
+
+                // Set the input state for an "input" port
+                var testInportData = "Test Input State 1";
+                var testInportLm = "LM1328113669.00000000000000000";
+                var initialTestState = { data: testInportData, 
+                                         lm: testInportLm };
+                var inportName = "input";
+
+                var inputContext = InputStates.call( component, 
+                                                     inportName, 
+                                                     initialTestState );
+
+                // Verify context returned is the component context 
+                verifyContext( inputContext );
+
+                // Verify the correct input state for input port 
+                var myInputState = InputStates.call( component, 
+                                                     inportName );
+                verifyInputState( myInputState, testInportData, testInportLm );
+
+                // Now change the port state to something else
+                testInportData = "A new test input state!";
+                testInportLm = "LM1328113661.00000000000000001";
+                var newTestState = { data: testInportData, 
+                                     lm: testInportLm };
+
+                inputContext = InputStates.call( component, 
+                                                 inportName, 
+                                                 newTestState );
+
+                // Verify the input state for input port is now the new state 
+                var newInputState = InputStates.call( component, 
+                                                      inportName );
+                verifyInputState( newInputState, testInportData, testInportLm );
+                verifyInputState( newInputState, testInportData, testInportLm );
```

Comment:
No.  newTestState is the value we fed into the setter.  I want to test the state we got back from the getter after calling the setter - that is going to be the newInputState. 

Logically what I wrote is correct, but clearly it is confusing.  I'll adjust this to have less confusing variable names.